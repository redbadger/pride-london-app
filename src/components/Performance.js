// @flow
import React from "react";
import { StyleSheet, View } from "react-native";
import formatDate from "date-fns/format";

import Text from "./Text";
import { lightishGreyColor } from "../constants/colors";

const timeFormat = "h:mma";

type Props = {
  title: string,
  startTime: string
};

const Performance = ({ title, startTime }: Props) => (
  <View style={styles.content}>
    <Text style={styles.startTime}>{formatDate(startTime, timeFormat)}</Text>
    <Text numberOfLines={3} style={styles.title}>
      {title}
    </Text>
  </View>
);

const styles = StyleSheet.create({
  content: {
    flexDirection: "row",
    padding: 12,
    borderBottomWidth: 1,
    borderColor: lightishGreyColor
  },
  startTime: {
    width: 90,
    fontWeight: "bold"
  },
  title: { flex: 1 }
});

export default Performance;
