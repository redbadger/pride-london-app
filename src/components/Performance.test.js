// @flow
import React from "react";
import { shallow } from "enzyme";
import Performance from "./Performance";

it("renders correctly", () => {
  const output = shallow(
    <Performance startTime="12:00" title="DJ Bag-o-Donuts" />
  );
  expect(output).toMatchSnapshot();
});
