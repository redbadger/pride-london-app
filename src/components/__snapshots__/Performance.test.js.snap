// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "paddingVertical": 12,
    }
  }
>
  <Text
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    style={
      Object {
        "fontWeight": "bold",
        "width": 90,
      }
    }
    type="text"
  >
    12:00
  </Text>
  <Text
    color="blackColor"
    markdown={false}
    markdownStyle={Object {}}
    numberOfLines={3}
    style={
      Object {
        "flex": 1,
      }
    }
    type="text"
  >
    DJ Bag-o-Donuts
  </Text>
</Component>
`;
